lib/Doctrine/DBAL/Driver/SQLAnywhere/SQLAnywhereConnection.php

```diff
@@ -137,13 +130,29 @@ public function getServerVersion()
     /**
      * {@inheritdoc}
      */
-    public function lastInsertId($name = null)
+    public function lastInsertId(?string $name = null) : string
     {
         if ($name === null) {
-            return sasql_insert_id($this->connection);
+            $result = sasql_insert_id($this->connection);
+
+            if ($result === false) {
+                throw SQLAnywhereException::fromSQLAnywhereError($this->connection);
+            }
+
+            if ($result === 0) {
+                throw new SQLAnywhereException('The last insert did not affect an AUTOINCREMENT column.');
+            }
+
+            return (string) $result;
         }
 
-        return $this->query('SELECT ' . $name . '.CURRVAL')->fetchColumn();
+        $result = $this->query('SELECT ' . $name . '.CURRVAL')->fetchColumn();
+
+        if ($result === false) {
+            throw new SQLAnywhereException('No sequence with name "' . $name . '" found.');
```

Comment:
Please use a factory method for consistency.

Reply:
Done.